data_diff/dbt.py

```diff
@@ -265,7 +266,11 @@ def _get_prod_path_from_manifest(model, prod_manifest) -> Union[Tuple[str, str,
     return prod_database, prod_schema, prod_alias
 
 
-def _local_diff(diff_vars: TDiffVars, json_output: bool = False) -> None:
+def _local_diff(
+    diff_vars: TDiffVars, json_output: bool = False, log_status_handler: Optional[LogStatusHandler] = None
+) -> None:
+    if log_status_handler:
+        log_status_handler.diff_started(diff_vars.dev_path[-1])
```

Comment:
In this method we're passing `diff_vars.threads` to connect_to_table, which is the `threads:` value from the dbt profile. This was \~okay\~ when running the diffs sequentially, but could greatly balloon the number of concurrent database threads when the outer diffs execute in parallel

Previously:
`n_concurrent_db_threads = diff_vars.threads`

PR as is:
`n_concurrent_db_threads = number_of_daemons * diff_vars.threads`

Running these in parallel, I think it makes more sense to use that value to handle the number of daemons running above:
```
diff_thread = run_as_daemon(_local_diff, diff_vars, json_output, log_status_handler)
diff_threads.append(diff_thread)
```

That would use the diff_vars.threads value more similarly to dbt. Having threads control the number of concurrent diffs.

Reply:
So you want me to refactor `connect_to_table` to NOT use `diff_vars.threads` anymore and bring that up a level so that `threads==number_of_daemons`?

Reply:
Removed `diff_vars.threads` param from `connect_to_table` and made it a param for max workers.
https://github.com/datafold/data-diff/pull/776/files#diff-8a5550cee0de299f3906d3b40b82bec419ce23c065b04923c3638906b21bda70R119